src/dos/dos.cpp

```diff
@@ -148,7 +150,52 @@ static uint16_t DOS_GetAmount(void) {
 #ifndef DOSBOX_CPU_H
 #include "cpu.h"
 #endif
-static inline void modify_cycles(Bits value) {
+
+// Taken from Dosbox-X
+int hdd_data_rate = 0;
+int fdd_data_rate = 0;
+
+void DOS_SetDataRate(int rate, DiskType type)
+{
+	if (type == DiskType::Floppy) {
+		hdd_data_rate = rate * BytesPerKilobyte;
+	} else {
+		fdd_data_rate = rate * BytesPerKilobyte;
+	}
+}
+
+void diskio_delay(Bits value /*bytes*/, DiskNoiseDevice* disknoise,
+                  DiskType type = DiskType::HardDisk)
+{
+	if ((type == DiskType::Floppy && fdd_data_rate != 0) ||
+	    (type != DiskType::Floppy && hdd_data_rate != 0)) {
```

Comment:
Using a magic value of `0` to signal the feature is disabled is not ideal. Please introduce separate boolean variables.